pilot/pkg/model/context.go

```diff
@@ -219,6 +224,30 @@ func ParseServiceNode(s string) (Proxy, error) {
 	return out, nil
 }
 
+// GetProxyConfigNamespace extracts the namespace associated with the proxy
+// from the proxy metadata or the proxy ID
+func GetProxyConfigNamespace(proxy *Proxy) string {
```

Comment:
nit: I see you use this function to set `ConfigNamespace`, if so I would name `func (node *Proxy) SetConfigNamespace() string` 

This can let user call it when need `ConfigNamespace`, thus I think no extra field needed in `Proxy struct`